src/components/Profile/Setting.vue

```diff
@@ -88,28 +93,70 @@ export default {
   components: {
     PreferenceItem
   },
+  data() {
+    return {
+      preference: {
+        isAnonymous: false,
+        editedNotification: false,
+        subscribedNotification: false }
+    }
+  },
   computed: {
     ...mapGetters(['displayName', 'photoURL']),
     ...mapGetters({ email: 'user/email' }),
     mockPreference() {
       return [
         new Preference(
+          'isAnonymous',
           '匿名發文',
           '開啟後您的新增段落與文章預設作者都將匿名',
-          false
+          this.preference.isAnonymous
         ),
         new Preference(
+          'editedNotification',
           '編輯文章更新通知',
           '開啟後您曾編輯過的文章有任何更新都將通知您',
-          false
+          this.preference.editedNotification
         ),
         new Preference(
+          'subscribedNotification',
           '收藏文章更新通知',
           '開啟後您的收藏文章有任何更新都將通知您',
-          false
+          this.preference.subscribedNotification
         )
       ]
     }
+  },
+  async mounted() {
+    try {
+      const { data } = await UserAPI.getPreference()
+      if (data.code === 200) {
+        this.preference = data.data
+      } else {
+        throw new Error(data.message)
+      }
+    } catch (e) {
+      console.error(e)
+      this.preference = {
+        isAnonymous: false,
+        editedNotification: false,
+        subscribedNotification: false }
+    }
```

Comment:
These code can be wrapped as a methods. (e.g. `fetchStatus()`

It will make the `mounted` hook function more readable if there are more tasks in the scope.

Reply:
done